aws-test/tests/aws_emr_cluster/variables.tf

```
@@ -0,0 +1,289 @@
+            "elasticmapreduce:ListSteps",
+            "kinesis:CreateStream",
+            "kinesis:DeleteStream",
+            "kinesis:DescribeStream",
+            "kinesis:GetRecords",
+            "kinesis:GetShardIterator",
+            "kinesis:MergeShards",
+            "kinesis:PutRecord",
+            "kinesis:SplitShard",
+            "rds:Describe*",
+            "s3:*",
+            "sdb:*",
+            "sns:*",
+            "sqs:*"
+        ]
+    }]
+}
+EOF
+}
+resource "aws_vpc" "my_vpc" {
+  cidr_block           = "168.31.0.0/16"
+}
+resource "aws_security_group" "emr_master" {
+  vpc_id                 = "${aws_vpc.my_vpc.id}"
+  revoke_rules_on_delete = true
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+  }
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  depends_on = ["aws_subnet.my_subnet"]
+  lifecycle {
+    ignore_changes = ["ingress", "egress"]
+  }
+}
+resource "aws_subnet" "my_subnet" {
+  vpc_id     = "${aws_vpc.my_vpc.id}"
+  cidr_block = "168.31.0.0/20"
+}
+resource "aws_internet_gateway" "gw" {
+  vpc_id = "${aws_vpc.my_vpc.id}"
+}
+resource "aws_route_table" "test_route_table" {
+  vpc_id = "${aws_vpc.my_vpc.id}"
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = "${aws_internet_gateway.gw.id}"
+  }
+}
+resource "aws_main_route_table_association" "test_association" {
+  vpc_id         = "${aws_vpc.my_vpc.id}"
+  route_table_id = "${aws_route_table.test_route_table.id}"
+}
+resource "aws_key_pair" "deployer" {
+  key_name   = "deployer-key"
+  public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC6B60dQX19dOO0wmNnPf27NJBXQhuVAsFcshY7YqTmNA1dfl0lXvJFSF0nVTvaxtE2854gmvygm/yafopG33zJeH+k2ZrblivkEuH0NNajKMp2mBumkk8RiTVvU1ET4m6Z1jZaK0dTSeV2k6gUk7Wmvo4+6RIu+wdAtwGGGcSq4HJ4M0G1CGHruBywMdOs5CklqRn7BRMh2aexowDachlabUKRFwTf7OCjDcoCWHo+o4kHV7NDEdhFzrE2fxXttt054kd5CMZOEOB9p01pAqSPrxFv3S/BpiXuhmjwsUsIdfHIgFrFhNY/M4+KhTa3pI69NqsigReotTEzk2QwHyHSjsRwfzeJNRgI8iqGIdYr3+t6zCKswyzgV0hqalZhXtpVUEBbumJAKWy+VzTnOXUQ5zEVvabTxi3dsW9zMGofjwhiC75x97RE7U/ZEanRM+0Avr8e3c3ob4NLf9JOwbvB/jYvS6j/nuwJ5H6igT3Oj0oDiknh2WtOgH/BKKaCq5Nyl15Df65i0PNgKRcLy+x0fubYUtsCOMPpQTfztSYGjbvAhAOrH2LJF0i2dTD3PcVw6R6uDpJSw9QkzdD+2ofxnqjGPUeKR3mraGOXt2hZe9F9bvvdZScDfdAoyBUYTO2/sxsQFobRtdiEk2up+mNIx9QXMXO2dOgtNUYZKc1DdQ== clara@turbot.com"
+}
+
+resource "aws_emr_cluster" "named_test_resource" {
+  name          = var.resource_name
+  release_label = "emr-4.9.5"
+  applications  = ["Spark"]
+  ec2_attributes {
+    key_name = "${aws_key_pair.deployer.key_name}"
+    subnet_id                         = "${aws_subnet.my_subnet.id}"
+    emr_managed_master_security_group = "${aws_security_group.emr_master.id}"
+    emr_managed_slave_security_group  = "${aws_security_group.emr_master.id}"
+    instance_profile                  = "${aws_iam_instance_profile.emr_profile.arn}"
+  }
+  master_instance_group {
+    instance_type = "m4.xlarge"
```

Can we create resource of lower configuration?
```suggestion
    instance_type = "m1.small"
```